test/core/derivatives/problem.py

```diff
@@ -28,22 +28,18 @@ def make_test_problems(settings):
 
 
 def add_missing_defaults(setting):
-    """Create derivative test problem from setting.
-
-    Args:
-        setting (dict): configuration dictionary
-
-    Returns:
-        DerivativesTestProblem: problem with specified settings.
-    """
-    required = ["module_fn", "module_kwargs", "input_kwargs"]
+    """Add missing entries in settings such that the new format works."""
+    required = ["module_fn", "input_fn"]
     optional = {
-        "input_fn": torch.rand,
+        "target_fn": lambda: None,
         "id_prefix": "",
+        "device": get_available_devices(),
         "seed": 0,
-        "target_fn": lambda: None,
+    }
+    overwrite = {
+        "module_kwargs": {},
+        "input_kwargs": {},
         "target_kwargs": {},
```

Comment:
The arguments `module_kwargs`, `input_kwargs`, and `target_kwargs` are legacy from the old format,not required by the new format. They should be removed. You will also have to refactor the implementation of `DerivativesTestProblem` below for that to work.